torch/CMakeLists.txt

```diff
@@ -323,11 +325,18 @@ target_link_libraries(torch caffe2_library)
 
 find_package(OpenMP)
 if(OPENMP_FOUND)
-  if (VERBOSE)
-    message(STATUS "Compiling with OpenMP")
+  message(STATUS "pytorch is compiling with OpenMP")
+  target_compile_options(torch INTERFACE ${OpenMP_CXX_FLAGS})
+  if (CMAKE_CXX_COMPILER_ID MATCHES "AppleClang")
+    #AppleClang 7+ should link the separated OpenMP lib.
+    #https://github.com/Kitware/CMake/blob/42212f7539040139ecec092547b7d58ef12a4d72/Modules/FindOpenMP.cmake#L252
+    if (CMAKE_CXX_COMPILER_VERSION VERSION_EQUAL "7.0" OR
+        CMAKE_CXX_COMPILER_VERSION VERSION_GREATER "7.0")
+      target_link_libraries(torch ${OpenMP_libomp_LIBRARY})
+    endif()
+  else()
+    target_link_libraries(torch -fopenmp)
```

Comment:
I think in general blindly linking `-fopenmp` is a bad idea. Considering the problem with `OpenMP_libomp_LIBRARY` being not available in 3.5 cmake, I would suggest port newer `FindOpenMP.cmake` over, and link against `OpenMP_<lang>_LIBRARIES`.

Reply:
Update to use ${OpenMP_CXX_LIBRARIES} now.